org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/ReorgCorrectionsSubProcessor.java

```diff
@@ -157,5 +166,24 @@ public static void removeImportStatementProposals(IInvocationContext context, IP
 				JavaLanguageServerPlugin.log(e);
 			}
 		}
+		List<CompilationUnitRewriteOperation> operations = new ArrayList<>();
+		for (IProblem compilationUnitProblem : context.getASTRoot().getProblems()) {
+			if (compilationUnitProblem.getID() == IProblem.UnusedImport) {
```

Comment:
Is there a reason to only do unused imports ? What about `UnusedCodeFixCore.isUnusedImport(IProblemLocationCore)` (it not only checks `IProblem.UnusedImport`, but some other cases like duplicate, conflicting, cannot import, not found)

Reply:
In fact, why not just use :

`UnusedCodeFixCore.createCleanUp(CompilationUnit)`. That should automatically handle all problems of the compilation unit and check if the import is unused. Only boolean to then set is `removeUnusedImports`

Reply:
> What about UnusedCodeFixCore.isUnusedImport(IProblemLocationCore) (it not only checks IProblem.UnusedImport

That should be. We should keep consistent with Quick Fix "remove unused import" in https://github.com/eclipse/eclipse.jdt.ls/blob/3823064062424dfd0ba89a3a9a6b6a278435ad87/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/QuickFixProcessor.java#L115-L121

> In fact, why not just use : UnusedCodeFixCore.createCleanUp(CompilationUnit)

You're right. Let's use this as it's a cleaner way.